src/Symfony/Bundle/SecurityBundle/DependencyInjection/MainConfiguration.php

```diff
@@ -218,10 +218,24 @@ private function addFirewallsSection(ArrayNodeDefinition $rootNode, array $facto
                 ->fixXmlConfig('delete_cookie')
                 ->children()
                     ->arrayNode('delete_cookies')
+                        ->normalizeKeys(false)
                         ->beforeNormalization()
                             ->ifTrue(function ($v) { return \is_array($v) && \is_int(key($v)); })
                             ->then(function ($v) { return array_map(function ($v) { return ['name' => $v]; }, $v); })
                         ->end()
+                        ->beforeNormalization()
+                            ->ifArray()->then(function ($v) {
+                                foreach ($v as $originalName => $config) {
+                                    if (false !== strpos($originalName, '-')) {
+                                        $normalizedName = str_replace('-', '_', $originalName);
+                                        $v[$normalizedName] = $config;
+                                        @trigger_error(sprintf('Normalization of cookie names is deprecated since Symfony 4.3. Starting from Symfony 5.0, only the "%s" cookie will be deleted, instead of both "%s" and "%s".', $originalName, $normalizedName, $originalName), E_USER_DEPRECATED);
```

Comment:
is `cookie-name` really deleted now? if normalization turns them to `cookie_name`, isn't only the later removed?

Reply:
This bug fix/feature will introduce the new behavior that deletes the 2 cookies. It's a bit strange ... but we're introducing a new feature and deprecating it at the same time. But maybe there's another way to solve this bug and provide a smooth upgrade.

Reply:
I'm not sure we need to delete the `-` version. Instead, we should ask ppl to rename the cookie in their config.

Reply:
I've changed some things according to your feedback. Thanks!